dev/run

```diff
@@ -571,8 +572,8 @@ def hashify(pwd, salt=COMMON_SALT, iterations=10, keylen=20):
     >>> hashify(candeira)
     -pbkdf2-99eb34d97cdaa581e6ba7b5386e112c265c5c670,d1d2d4d8909c82c81b6c8184429a0739,10
     """
-    derived_key = pbkdf2_hex(pwd, salt, iterations, keylen)
-    return "-pbkdf2-%s,%s,%s" % (derived_key, salt, iterations)
+    derived_key = pbkdf2_hex(pwd, salt, iterations, keylen, hashfunc=hashlib.sha256)
+    return "-pbkdf2:sha256-%s,%s,%s" % (derived_key, salt, iterations)
```

Comment:
I couldn't start a local dev n=1 node cluster for some reason.

```
 % ./dev/run --admin=adm:pass -n 1
[ * ] Setup environment ... ok
[ * ] Ensure CouchDB is built ... ok
[ * ] Ensure Erlang boot script exists ... ok
[ * ] Prepare configuration files ... ok
[ * ] Start node node1 ... ok
[ * ] Check node at http://127.0.0.1:15984/ ... ok
[ * ] Running cluster setup ... failed: b'{"error":"unauthorized","reason":"Name or password is incorrect."}\n'
Traceback (most recent call last):
  File "/Users/nvatama/asf/./dev/run", line 934, in <module>
    main()
  File "/Users/nvatama/asf/./dev/run", line 93, in main
    startup(ctx)
  File "/Users/nvatama/asf/./dev/run", line 593, in startup
    cluster_setup(ctx)
  File "/Users/nvatama/asf/./dev/run", line 73, in wrapper
    res = func(*args, **kwargs)
  File "/Users/nvatama/asf/./dev/run", line 743, in cluster_setup
    finish_cluster(lead_port, *ctx["admin"])
  File "/Users/nvatama/asf/./dev/run", line 829, in finish_cluster
    assert resp.status in (201, 400), resp.read()
AssertionError: b'{"error":"unauthorized","reason":"Name or password is incorrect."}\n'
```

The logs showed:

```
[notice] 2023-11-07T23:19:16.524311Z node1@127.0.0.1 <0.494.0> -------- Missing system database _users
[notice] 2023-11-07T23:19:17.254909Z node1@127.0.0.1 <0.430.0> e86161454e 127.0.0.1:15984 127.0.0.1 undefined GET / 200 ok 63
[notice] 2023-11-07T23:19:17.276089Z node1@127.0.0.1 <0.431.0> acf614ed19 Setup: [{<<"action">>,<<"enable_cluster">>},{<<"bind_address">>,<<"0.0.0.0">>},{<<"username">>,<<"adm">>},{<<"password">>,<<"****">>},{<<"node_count">>,1}]

[notice] 2023-11-07T23:19:17.278581Z node1@127.0.0.1 <0.141.0> -------- config: [admins] adm set to '****' for reason nil
[notice] 2023-11-07T23:19:17.281576Z node1@127.0.0.1 <0.141.0> -------- config: [chttpd] bind_address set to 0.0.0.0 for reason nil
[notice] 2023-11-07T23:19:17.282793Z node1@127.0.0.1 <0.141.0> -------- config: [cluster] n set to 1 for reason nil
[error] 2023-11-07T23:19:17.282809Z node1@127.0.0.1 <0.427.0> -------- Supervisor chttpd_sup had child chttpd started with chttpd:start_link() at <0.429.0> exit with reason normal in context child_terminated
[error] 2023-11-07T23:19:17.282840Z node1@127.0.0.1 <0.427.0> -------- Supervisor chttpd_sup had child chttpd started with chttpd:start_link() at <0.429.0> exit with reason normal in context child_terminated
[notice] 2023-11-07T23:19:17.284662Z node1@127.0.0.1 <0.431.0> acf614ed19 127.0.0.1:15984 127.0.0.1 adm POST /_cluster_setup 201 ok 26
[notice] 2023-11-07T23:19:17.286358Z node1@127.0.0.1 <0.585.0> 53d7202075 127.0.0.1:15984 127.0.0.1 undefined POST /_cluster_setup 401 ok 0
[info] 2023-11-07T23:19:17.288890Z node1@127.0.0.1 <0.66.0> -------- SIGTERM received - shutting down
[info] 2023-11-07T23:19:17.288923Z node1@127.0.0.1 <0.66.0> -------- SIGTERM received - shutting down
```
I couldn't yet tell why though. I'll try adding more debug statements in there to figure it out

Reply:
`-n 3` seems to work though, so it's a bit strange.

Reply:
one note here: an admin cred can't be used until it's hashed, so if iterations is high (or the machine is slow), dev/run doesn't boot as it doesn't wait long enough for the hashing. This is a latent issue, it's just exposed more with the new defaults (50_000 in default.ini, which is why the Makefile overrides it to 9, so those dev/run's boot up quickly).

Reply:
That explains it, thanks! As part of that setup we re-set the admin credentials by calling setup.erl [set_admin](https://github.com/apache/couchdb/blob/main/src/setup/src/setup.erl#L172). That sets it to plaintext password and then immediately returns. The config:set then asynchronously kicks of the couch_server admin re-hasher.